.github/workflows/ci.yml

```diff
@@ -0,0 +1,121 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - develop
+      - backports/**
+  pull_request:
+    branches:
+      - develop
+      - backports/**
+    paths-ignore:
+      - '**.rst'
+      - '**.md'
+      - 'scripts/dev'
+  merge_group:
+    types: [checks_requested]
+
+concurrency:
+  group: ${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
+  cancel-in-progress: true
```

Comment:
This settings means that for all workflow group, a new items will cancel the previous on going or queue runs.  For pull-request this is clearly what we want.  For push in the repository it is more debatable; having the setting on for those means that technically not all pushes are tested/processed.

Reply:
Good point; how can we set that up to apply only to pull requests?

Reply:
We  will need two (or 3) yml files/workflow and use '[reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)`

Something like:
```
name: ci_push

on:
  push:
    branches:
      - develop
      - backports/**

  jobs:
  call-workflow:
    uses: .github/workflows/ci.yml
```
and `ci.yml` will replace the push section with: ```workflow_call:``` section.

Reply:
Actually are you sure it will cancel the runs on `develop`? I think it'll cancel only other jobs in the same group, and if it's not a pull request, then it'll use the unique run number.
```
  group: ${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
```

Reply:
I am not sure, this might indeed be sufficient if I read the doc for `github.run_number` correctly.

Reply:
Yeah, I've got a draft of this in #1021